docs/topics/tasks.txt

```diff
@@ -0,0 +1,335 @@
+module-level function::
+
+    from django.tasks import task
+
+
+    @task()
+    def calculate_meaning_of_life(answer=42):
+        return answer
+
+Returned in a :class:`django.tasks.Task` instance.
+
+.. note::
+
+    All arguments and return values must be JSON serializable, and round-trip
+    as the same types.
+
+    This means complex types like model instances, as well as many built-in types
+    like ``datetime`` and ``tuple`` cannot be used in tasks without additional
+    conversion.
+
+The ``task`` decorator accepts a few keyword arguments to customize the task:
+
+* ``priority``: The priority of the task. Higher numbers will be executed sooner.
+
+* ``queue_name``: The name of the queue the task will be executed on
+
+* ``backend``: The name of the backend this task must use (as defined in
+  :setting:`TASKS`).
+
+* ``enqueue_on_commit``: Whether the task is enqueued when the current transaction
+  commits successfully, or enqueued immediately.
+
+These arguments correspond to attributes on the created
+:class:`Task <django.tasks.Task>`.
+
+By convention, tasks should be defined in a ``tasks.py`` file, however this is
+not enforced.
+
+Modifying tasks
+---------------
+
+Before enqueueing tasks, it may be necessary to modify certain parameters of the task.
+For example, to give it a higher priority than it would normally.
+
+A ``Task`` instance cannot be modified directly. Instead, a modified instance
+can be created with the :meth:`using <django.tasks.Task.using>` method, leaving
+the original as-is::
+
+    calculate_meaning_of_life.priority  # 0
+
+    calculate_meaning_of_life.using(priority=10).priority  # 10
+
+``using`` allows modifying the following attributes:
+
+* :attr:`priority <django.tasks.Task.priority>`
+* :attr:`backend <django.tasks.Task.backend>`
+* :attr:`queue_name <django.tasks.Task.queue_name>`
+* :attr:`run_after <django.tasks.Task.run_after>`
+
+``run_after`` may also be provided as a :class:`timedelta <datetime.timedelta>`,
+which is used relative to the current time (when ``using`` is called), or a
+timezone-aware :class:`datetime <datetime.datetime>`.
+
+Enqueueing tasks
+================
+
+To add the task to the queue store, so it will be executed, call the ``enqueue``
+method on it::
+
+    result = calculate_meaning_of_life.enqueue()
+
+Returned is a :class:`django.tasks.TaskResult`, which can be used to retrieve
+the result of the task once it has finished executing.
+
+If the task takes arguments, these can be passed as-is to ``enqueue``::
+
+    result = calculate_meaning_of_life.enqueue(answer=42)
+
+To enqueue tasks in an ``async`` context, :meth:`aenqueue <django.tasks.Task.aenqueue>`
+is available as an ``async`` variant of ``enqueue``.
+
+Transactions
+------------
+
+By default, tasks are enqueued after the current transaction (if there is one)
+commits successfully (using :meth:`transaction.on_commit <django.db.transaction.on_commit>`),
+rather than enqueueing immediately.
```

Comment:
```suggestion
rather than enqueueing immediately.
This is the default because often a task depends on an object that is being stored in the database,
and you want to ensure the object is present before the task runs. 
```

I think this explains why this is the default?